### ISSUE TITLE
Problems in the Release Process for Auto-documentation Modules

### ISSUE THREAD

**Issue by PlagueHO** [PowerShell/DscResources#315]
The Auto-documentation modules xNetworking, xCertificate, xStorage, xDFS that are being published to the PS Gallery are missing the `en-US` folder, the `README.MD`, `LICENSE` and `CHANGELOG.MD` files. It is also including the `README.MD` files in each resource which it should not.

The Auto-documentation module SharePointDsc that is being published to the PS Gallery is missing the `README.MD`, `LICENSE` and `CHANGELOG.MD` files. It is also including the `README.MD` files in each resource which it should not.

The NUPKG files that are produced by the AppVeyor build do contain the `en-US` folder and have the README.MD files removed (see https://github.com/PowerShell/DscResource.Tests/blob/dev/AppVeyor.psm1#L416).

@kwirkykat - is it possible that the release process that is used to publish the modules to the PS Gallery is slightly different to what happens during the AppVeyor CI? I'm not sure why SharePointDsc includes the `en-us` but the xNetworking, xCertificate, xStorage and xDFS does not- any help here would be greatly appreciated. I'd definitely like to also include the `README.MD`, `CHANGELOG.MD` and `LICENSE` in file published to the PS Gallery as well.

I have raised a separate issue to ensure the `README.MD`, `CHANGELOG.MD` and `LICENSE` is included in the ZIP and NUPKG files produced in the AppVeyor CI process [here](https://github.com/PowerShell/DscResource.Tests/issues/179) - I'll submit a PR to resolve this issue.

**Comment by PlagueHO** [comment]
Tagging @BrianFarnhill as this also affects SharePointDsc so he might have some ideas.

**Comment by BrianFarnhill** [comment]
@PlagueHO So the reason for this is that as far as I know, the process that @kwirkykat uses (at least last time I spoke to her about it, which was a while ago) takes the source code from GitHub rather than the output of the build from AppVeyor (its why you'll see the release number for SharePoint are all 1.x.0.0 instead of having a build number on them). We manually copy the contents of the en-US folder for the documentation in to the master branch when we merge there so that it gets included. I agree that having it pull from the AppVeyor builds would be a better outcome, but thats up to @kwirkykat since she knows whats involved in the publishing procedure - I'm happy to be a team player and do whatever is easy for her at this point.

**Comment by PlagueHO** [comment]
Thanks for clarifying that @BrianFarnhill! The release process has been a bit of a mystery, so that explains why the files aren't making it into the package.

I definitely don't want to make any additional processes for @kwirkykat.

So, how do you get the en-US folder migrated into the master branch at release time? If we could get that to happen for xNetworking, xStorage, xCertificate and xDFS then this would mostly solve my issue. It would still be good to copy the README.MD, CHANGELOG.MD and LICENSE files copied in at the same time as well - this would result in a much more complete set of information being provided in the package for these resources.

**Comment by BrianFarnhill** [comment]
@PlagueHO I dont know about copying the readme and changelog in to the module itself - the license is probably a good call though. 

The process I take is like this:

1. Create a "realease-1.x" branch based on dev
2. Pull master in to dev (to resolve any merge conflicts)
3. Update the contents of the en-US folder to match what would go in to the build
4. Merge the release branch in to dev (I leave the PR for this open for @kwirkykat so she can do the release at the same time as finalising the PR, thats how she knows we have a new build ready to go)

**Comment by PlagueHO** [comment]
Ah! Thanks @BrianFarnhill - that explains a lot! We're not using any of these additional steps in releasing xNetworking etc. So perhaps we need to as well.

But I see how things work now - I didn't notice that you had an en-US folder in the actual repo itself. I just made the assumption that it was being built into the package during release time.

I understand why you don't want to copy in the README.MD and CHANGELOG.MD: So that you don't end up with two copies in the repo.

@KwirkyKat is this a process we could look at using for xNetworking, xStorage, xCertificate and xDFS (and soon xComputerManagement)? Does this add any additional overhead for you if we do this?

@tysonjhayes - do you have an opinion on this one?

**Comment by kwirkykat** [comment]
@PlagueHO It doesn't create any extra overhead for me, but it will create extra overhead for you. If you want to open your own release PRs that's fine with me, but then I need to know what to do if a module has changes and you haven't opened a release PR. For SharePointDsc Brian takes care of all of it, so I don't release that module even if it has changes unless he has opened a release PR.

This is what happens when a module is released:
1. Run the first release script. This performs all the following actions:
  - Checks out the latest code from the dev branch on GitHub
  - Inserts a new version section under the 'Unreleased' header in either README.md or CHANGELOG.md in the root folder 
  - Updates the version in the module manifest
  - Finds the version release notes by the correct header in either README.md or CHANGELOG.md in the root folder, replaces any single quote characters with double quotes and removes any pound/hash characters, and then inserts the release notes into the ReleaseNotes property in the module manifest
  - Commits and pushes to the dev branch on GitHub
  - Opens a release pull request to the master branch
2. Wait until all tests have passed on the opened PR and check that all the correct changes were made
3. Merge the release PR
4. Tag the release commit in the master branch
5. Run the second release script. This performs all the following actions:
  - Checks out the latest code from the master branch on GitHub
  - Calls Publish-Module on the module manifest

I don't touch the package that AppVeyor creates.
@BrianFarnhill does step 1 on his own since he creates a separate release branch so I just go to step 2 for SharePointDsc. I don't mind having the separate release branch. That's probably the more proper way to release anyways.

**Comment by PlagueHO** [comment]
Thanks @kwirkykat - that is +++ helpful. We may need to adopt the same process for xNetworking, xStorage, xCertificate, xDFS if you're OK with this? E.g. Myself or @tysonjhayes will need to create the Release branch just prior to the monthly release.

**Comment by kwirkykat** [comment]
@PlagueHO Fine by me 👍

**Comment by kwirkykat** [comment]
@PlagueHO There's a release tomorrow, and xNetworking has changes that need to go out. Do you want me to release per usual or wait for you to create a release branch?

**Comment by PlagueHO** [comment]
Hi @kwirkykat - I'd like to try the release branch. Will chat on the call in 45 mins if that is ok?

**Comment by kwirkykat** [comment]
@PlagueHO Yes will chat in the call.
Also release is not tomorrow. Don't know why I was thinking that all of a sudden... It's next Wednesday (Oct 4). So you have time.

**Comment by PlagueHO** [comment]
Haha no worries! :smile: that is even better as I'm trying really hard to get the new xProxySettings resource into xNetworking before release. But I'm still keen to try the release branch mechanism for xNetworking when it happens. I'll put this in my calendar as a reminder.

**Comment by PlagueHO** [comment]
Hi @kwirkykat - I've prepared the release branch for xNetworking according to @BrianFarnhill's instructions:
https://github.com/PowerShell/xNetworking/tree/Release-5.2.0.0

I assumed the next release version will be 5.2.0.0. If I've mucked up the process, feel free to ignore/delete the release branch and release as normal - but let me know what I messed up so I can try again next time.

Thank you both for helping me to work through this!
